pkg/kn/commands/service/update.go

```diff
@@ -92,48 +88,46 @@ func NewServiceUpdateCommand(p *commands.KnParams) *cobra.Command {
 				}
 				err = editFlags.Apply(service, baseRevision, cmd)
 				if err != nil {
-					return err
+					return nil, err
 				}
 
 				if trafficFlags.Changed(cmd) {
 					traffic, err := traffic.Compute(cmd, service.Spec.Traffic, &trafficFlags)
 					if err != nil {
-						return err
+						return nil, err
 					}
 
 					service.Spec.Traffic = traffic
 				}
+				return service, nil
+			}
+
+			// Do the actual update with retry in case of conflicts
+			err = client.UpdateServiceWithRetry(name, updateFunc, MaxUpdateRetries)
+			if err != nil {
+				return err
+			}
 
-				err = client.UpdateService(service)
+			out := cmd.OutOrStdout()
+			//TODO: deprecated condition should be once --async is gone
+			if !waitFlags.Async && !waitFlags.NoWait {
+				fmt.Fprintf(out, "Updating Service '%s' in namespace '%s':\n", args[0], namespace)
+				fmt.Fprintln(out, "")
+				err := waitForService(client, name, out, waitFlags.TimeoutInSeconds)
 				if err != nil {
-					// Retry to update when a resource version conflict exists
-					if apierrors.IsConflict(err) && retries < MaxUpdateRetries {
-						retries++
-						continue
-					}
 					return err
 				}
-
-				out := cmd.OutOrStdout()
-				//TODO: deprecated condition should be once --async is gone
-				if !waitFlags.Async && !waitFlags.NoWait {
-					fmt.Fprintf(out, "Updating Service '%s' in namespace '%s':\n", args[0], namespace)
-					fmt.Fprintln(out, "")
-					err := waitForService(client, name, out, waitFlags.TimeoutInSeconds)
-					if err != nil {
-						return err
-					}
-					fmt.Fprintln(out, "")
-					return showUrl(client, name, latestRevisionBeforeUpdate, "updated", out)
-				} else {
-					if waitFlags.Async {
-						fmt.Fprintf(out, "\nWARNING: flag --async is deprecated and going to be removed in future release, please use --no-wait instead.\n\n")
-					}
-					fmt.Fprintf(out, "Service '%s' updated in namespace '%s'.\n", args[0], namespace)
+				fmt.Fprintln(out, "")
+				return showUrl(client, name, latestRevisionBeforeUpdate, "updated", out)
+			} else {
+				if waitFlags.Async {
+					fmt.Fprintf(out, "\nWARNING: flag --async is deprecated and going to be removed in future release, please use --no-wait instead.\n\n")
```

Comment:
Isn't this renamed to --no-wait with #438

Reply:
Yes, but we still accept `--async` even it is deprecated.

The deprecation process is supposed to be:
* Support both `--no-wait` and `--async` for the next n releases (n == 4 for knative/serving, but I think we could be faster). 
* Print a warning if the user uses a deprecated, with a hint to the replacement.